sdk/subscribe_block.go

```diff
@@ -24,6 +25,90 @@ const (
 	pathCosignature        = "cosignature"
 )
 
+// Closes the subscription channel.
+func closeChannel(s *subscribe) {
+	switch s.Ch.(type) {
+	case chan *BlockInfo:
+		chType := s.Ch.(chan *BlockInfo)
+		close(chType)
+
+	case chan *StatusInfo:
+		chType := s.Ch.(chan *StatusInfo)
+		delete(statusInfoChannels, s.getAdd())
+		close(chType)
+
+	case chan *HashInfo:
+		chType := s.Ch.(chan *HashInfo)
+		delete(unconfirmedRemovedChannels, s.getAdd())
+		close(chType)
+
+	case chan *PartialRemovedInfo:
+		chType := s.Ch.(chan *PartialRemovedInfo)
+		delete(partialRemovedInfoChannels, s.getAdd())
+		close(chType)
+
+	case chan *SignerInfo:
+		chType := s.Ch.(chan *SignerInfo)
+		delete(signerInfoChannels, s.getAdd())
+		close(chType)
+
+	default:
+		chType := s.Ch.(chan Transaction)
+		if s.getSubscribe() == "partialAdded" {
+			delete(partialAddedChannels, s.getAdd())
+		} else if s.getSubscribe() == "unconfirmedAdded" {
+			delete(unconfirmedAddedChannels, s.getAdd())
+		} else {
+			delete(confirmedAddedChannels, s.getAdd())
+		}
+		close(chType)
+	}
+}
+
+// Unsubscribe terminates the specified subscription.
+// It does not have any specific param.
+func (c *subscribe) unsubscribe() error {
```

Comment:
This method panics:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1454c17]
goroutine 28 [running]:
golang.org/x/net/websocket.Codec.Send(0x16dbbb8, 0x16dbbc0, 0x0, 0x1538ca0, 0xc0003570e0, 0x0, 0x0)
	/go/src/golang.org/x/net/websocket/websocket.go:313 +0xa7
github.com/proximax-storage/proximax-nem2-sdk-go/sdk.(*subscribe).unsubscribe(0xc00028cd80, 0x0, 0xc0002aeb38)
	/go/src/github.com/proximax-storage/proximax-nem2-sdk-go/sdk/subscribe_block.go:71 +0xc0
github.com/proximax-storage/proximax-nem2-sdk-go/sdk.(*SubscribeStatus).Unsubscribe(0xc0003260e0, 0xc00021fac0, 0x40)
	/go/src/github.com/proximax-storage/proximax-nem2-sdk-go/sdk/websocket.go:349 +0x2e
github.com/proximax-storage/go-xpx/blockchain.(*BlockChain).AnnounceFromAccount.func1(0xc0001bc550, 0xc0002719e0, 0xc0002aeae0, 0xc00027ebd0, 0x1744780, 0xc0000aa008)
	/go/src/github.com/proximax-storage/go-xpx/blockchain/blockchain.go:241 +0x2aa
created by github.com/proximax-storage/go-xpx/blockchain.(*BlockChain).AnnounceFromAccount
	/go/src/github.com/proximax-storage/go-xpx/blockchain/blockchain.go:210 +0x179

Reply:
fixed up!